inductiva/simulators/simulator.py

```diff
@@ -36,6 +39,15 @@ def _setup_input_dir(self, input_dir: types.Path):
                 f"The provided path (\"{input_dir}\") is not a directory.")
         return input_dir
 
+    def _check_resource_pool(self, machine_group):
+        if not self.checked_resource_pool:
+            self.checked_resource_pool = True
+            if inductiva.resources.machine_groups.get(
```

Comment:
This get should be the last thing you do, since otherwise you will call the API unnecessarily, for e.g. when machine_group is None, which is exactly when you want to check it.